oneview_redfish_toolkit/tests/blueprints/test_computer_system.py

```diff
@@ -193,3 +198,131 @@ def test_get_computer_system(self, mock_get_ov_client):
         self.assertEqual(computer_system_str, json_str)
+
+    @mock.patch.object(util, 'get_oneview_client')
+    def test_change_power_state(self, mock_get_ov_client):
+        """Tests change SH power state with valid power value"""
+
+        # Loading server_hardware mockup value
+        with open(
+                'oneview_redfish_toolkit/mockups_oneview/ServerHardware.json'
+        ) as f:
+            sh_dict = json.load(f)
+
+        # Loading ServerHardwareTypes mockup value
+        with open(
+                'oneview_redfish_toolkit/mockups_oneview/'
+                'ServerHardwareTypes.json'
+        ) as f:
+            sht_dict = json.load(f)
+
+        ov = mock_get_ov_client()
+        ov.server_hardware.get.return_value = sh_dict
+        ov.server_hardware_types.get.return_value = sht_dict
+        ov.server_hardware.update_power_state.return_value = {"status": "OK"}
+
+        reset_types = ["On", "ForceOff", "GracefulShutdown",
+                       "GracefulRestart", "ForceRestart", "PushPowerButton"]
+
+        for reset_type in reset_types:
+            response = self.app.post("/redfish/v1/Systems/30303437-3034"
+                                     "-4D32-3230-313133364752/Actions/"
+                                     "ComputerSystem.Reset",
+                                     data=dict(ResetType=reset_type))
+
+            # Tests response
+            self.assertEqual(status.HTTP_200_OK, response.status_code)
+            self.assertEqual("application/json", response.mimetype)
+
+            json_str = response.data.decode("utf-8")
+
+            self.assertEqual(json_str, '{"ResetType": "%s"}' % reset_type)
+
+    @mock.patch.object(util, 'get_oneview_client')
+    def test_change_power_state_invalid_value(self, mock_get_ov_client):
+        """Tests change SH power state with invalid power value"""
+
+        # Loading server_hardware mockup value
+        with open(
+                'oneview_redfish_toolkit/mockups_oneview/ServerHardware.json'
+        ) as f:
+            sh_dict = json.load(f)
+
+        # Loading ServerHardwareTypes mockup value
+        with open(
+                'oneview_redfish_toolkit/mockups_oneview/'
+                'ServerHardwareTypes.json'
+        ) as f:
+            sht_dict = json.load(f)
+
+        ov = mock_get_ov_client()
+        ov.server_hardware.get.return_value = sh_dict
+        ov.server_hardware_types.get.return_value = sht_dict
+
+        response = self.app.post("/redfish/v1/Systems/30303437-3034-4D32-3230"
+                                 "-313133364752/Actions/ComputerSystem.Reset",
+                                 data=dict(ResetType="INVALID_TYPE"))
+
+        # Tests response
+        self.assertEqual(status.HTTP_400_BAD_REQUEST, response.status_code)
+        self.assertEqual("application/json", response.mimetype)
+
+    @mock.patch.object(util, 'get_oneview_client')
+    def test_change_power_state_unexpected_error(self, mock_get_ov_client):
```

Comment:
How different are `test_change_power_state_unexpected_error` and `test_change_power_state_sh_exception`?

Reply:
I think it's possible to occur a different Exception of HPOneViewException, so `test_change_power_state_unexpected_error` tests general Exception and `test_change_power_state_sh_exception` only the HPOneViewException.